src/utils.cc

```diff
@@ -7,16 +7,16 @@
 
 namespace utils {
 
-    bool compile(std::string compiler, std::string flags, std::string src, std::string out) {
+    bool compile(const std::string& compiler, const std::string& flags, const std::string& src, const std::string& out) {
         /// TODO: check if the files need to be compiled or not
 
-        std::string args =  std::string(src) + " -c " + flags + " -o " + out;
+        const std::string& args =  std::string(src) + " -c " + flags + " -o " + out;
```

Comment:
This doesn't seem correct as a const reference. It should be just a simple std::string.